<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="pmcbioc">
<title>Introduction to 'pmcbioc' • pmcbioc</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to 'pmcbioc'">
<meta property="og:description" content="pmcbioc">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">pmcbioc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/introduction.html">Introduction to 'pmcbioc'</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/mtmorgan/pmcbioc/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="introduction_files/htmlwidgets-1.6.2/htmlwidgets.js"></script><link href="introduction_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="introduction_files/datatables-binding-0.28/datatables.js"></script><link href="introduction_files/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="introduction_files/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="introduction_files/dt-core-1.13.4/js/jquery.dataTables.min.js"></script><link href="introduction_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="introduction_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to 'pmcbioc'</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/pmcbioc/blob/HEAD/vignettes/introduction.Rmd" class="external-link"><code>vignettes/introduction.Rmd</code></a></small>
      <div class="d-none name"><code>introduction.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="installation-and-loading">Installation and loading<a class="anchor" aria-label="anchor" href="#installation-and-loading"></a>
</h2>
<p>Install the development version of <a href="https://mtmorgan.github.io/pmcbioc/">pmcbioc</a> from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"mtmorgan/pmcbioc"</span><span class="op">)</span></span></code></pre></div>
<p>Load the package with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mtmorgan.github.io/pmcbioc/">pmcbioc</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="retrieval-and-processing-of-pubmedcentral-data">Retrieval and processing of PubMedCentral data<a class="anchor" aria-label="anchor" href="#retrieval-and-processing-of-pubmedcentral-data"></a>
</h2>
<p><a href="https://mtmorgan.github.io/pmcbioc/">pmcbioc</a> requires
that the user visit the <a href="https://pubmed.ncbi.nlm.nih.gov/" class="external-link">PubMedCentral</a> web site,
search for terms or citations of intersted, and chooses to save the
results to their local disk as an XML file. The XML contains <em>full
text</em> of all articles satisfying the query, which means that it can
be quite large; 71,259 citations to the term ‘Bioconductor’ retrieved on
2023-09-13 occupy about 8.5GB of disk space. Suppose that the result
file is <code>pmc_result.xml</code>, in the current working
directory.</p>
<p>Data processing involves four steps</p>
<ol style="list-style-type: decimal">
<li>Create a <a href="https://duckdb.org/" class="external-link">DuckDB</a> database, to be
accessed using the <a href="https://cran.r-project.org/package=duckdb" class="external-link">duckdb</a> <em>R</em>
package.</li>
<li>Parse article metadata (PubMedCentral id, title, journal, year,
pmid, authors, keywords, citation PubMed ids) into tables in the
database.</li>
<li>Index the XML file to allow fast access to the full record of
individual articles, and record the index in the database.</li>
<li>Closing and perhaps renaming the database.</li>
</ol>
<div class="section level3">
<h3 id="database-creation">Database creation<a class="anchor" aria-label="anchor" href="#database-creation"></a>
</h3>
<p>Create a new DuckDB database in the current working directory. Use a
temporary file name to avoid accidentally over-writing existing
work.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">db_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>tmpdir <span class="op">=</span> <span class="st">"."</span>, fileext <span class="op">=</span> <span class="st">".duckdb"</span><span class="op">)</span></span>
<span><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu">pmcbioc_db</span><span class="op">(</span><span class="va">db_file</span><span class="op">)</span></span>
<span><span class="va">db</span></span>
<span><span class="co">#&gt; pmcbioc_db: ./fileda48126e5bcb.duckdb</span></span>
<span><span class="co">#&gt; connected: TRUE</span></span>
<span><span class="co">#&gt; read_only: FALSE</span></span>
<span><span class="co">#&gt; db_tables():</span></span></code></pre></div>
<p>Connection to a new (empty) database allows for updates
(<code>read_only: FALSE</code>). Connecting to an existing database is
by defualt read-only (<code>read_only: TRUE</code>); an existing
database can be opened to allow updates with the
<code>pmcbioc_db()</code> argument <code>read_only = TRUE</code>.</p>
</div>
<div class="section level3">
<h3 id="article-metadata">Article metadata<a class="anchor" aria-label="anchor" href="#article-metadata"></a>
</h3>
<p>Parse metadata about each article in the XML file into the database
using <code><a href="../reference/xml_parse.html">xml_parse()</a></code>. The arguments are the path to the XML
file, and the database created in the previous step.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## not evaluated</span></span>
<span><span class="fu"><a href="../reference/xml_parse.html">xml_parse</a></span><span class="op">(</span><span class="st">"pmc_result.xml"</span>, <span class="va">db</span><span class="op">)</span></span></code></pre></div>
<p>This process is expensive in both time and memory. The Bioconductor
query took about 1 minute per 1000 records, and ultimately consumed
about 18Gb of memory. Processing using
<code><a href="https://rdrr.io/pkg/XML/man/xmlEventParse.html" class="external-link">XML::xmlEventParse()</a></code>, with ‘<a href="https://www.w3.org/TR/1999/REC-xpath-19991116/" class="external-link">XPath</a>’ queries
used to extract metadata from each article. Details of the paths used
are visble using the (non-exported) function</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## not evaluated</span></span>
<span><span class="fu">pmcbioc</span><span class="fu">:::</span><span class="fu">xml_branches</span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span></code></pre></div>
<p>When the process completes, the database will contain four tables</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">db_tables</span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "article" "author"  "keyword" "refpmid"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="xml-index">XML index<a class="anchor" aria-label="anchor" href="#xml-index"></a>
</h3>
<p>The function <code><a href="../reference/xml_index.html">xml_index()</a></code> creates an index on the large
XML file downloaded from PubMedCentral, making it easy to query subsets
of articles for information not extracted to the database with
<code><a href="../reference/xml_parse.html">xml_parse()</a></code>. Create and add the index with</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## not evaluated</span></span>
<span><span class="fu"><a href="../reference/xml_index.html">xml_index</a></span><span class="op">(</span><span class="st">"pmc_result.xml"</span>, <span class="va">db</span><span class="op">)</span></span></code></pre></div>
<p>This process is faster and less memory-intensive than
<code><a href="../reference/xml_parse.html">xml_parse()</a></code>, but still takes several minutes. On
completion, the database contains an additional table
<code>index</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">db_tables</span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "article" "author"  "index"   "keyword" "refpmid"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="final-steps">Final steps<a class="anchor" aria-label="anchor" href="#final-steps"></a>
</h3>
<p>Remember to disconnect from the database, so any buffered data is
written to disk.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">db_disconnect</span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span><span class="co">#&gt; pmcbioc_db: ./fileda48126e5bcb.duckdb</span></span>
<span><span class="co">#&gt; connected: FALSE</span></span>
<span><span class="co">#&gt; read_only: FALSE</span></span></code></pre></div>
<p>Optionally, rename the temporary file to something permanent.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## not evaluated</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.rename</a></span><span class="op">(</span><span class="va">db_file</span>, <span class="st">"pmc_result.duckdb"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="exploring-our-data">Exploring our data<a class="anchor" aria-label="anchor" href="#exploring-our-data"></a>
</h2>
<p>Start by connecting to the database. Since the database already
exists, it opens as ‘read only’ so one does not have to worry about
accidentally corrupting it.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu">pmcbioc_db</span><span class="op">(</span><span class="st">"~/a/github/pmcbioc/pmc_result.duckdb"</span><span class="op">)</span></span>
<span><span class="va">db</span></span>
<span><span class="co">#&gt; pmcbioc_db: ~/a/github/pmcbioc/pmc_result.duckdb</span></span>
<span><span class="co">#&gt; connected: TRUE</span></span>
<span><span class="co">#&gt; read_only: TRUE</span></span>
<span><span class="co">#&gt; db_tables(): article, author, index, keyword, refpmid</span></span></code></pre></div>
<div class="section level3">
<h3 id="article-metadata-1">Article metadata<a class="anchor" aria-label="anchor" href="#article-metadata-1"></a>
</h3>
<div class="section level4">
<h4 id="article">
<code>article</code><a class="anchor" aria-label="anchor" href="#article"></a>
</h4>
<p>Use <a href="https://cran.r-project.org/package=dplyr" class="external-link">dplyr</a>
functions to open and explore tables. Start with the
<code>article</code> table.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"article"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   table&lt;article&gt; [?? x 6]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB 0.8.1 [root@Darwin 21.6.0:R 4.3.1/~/a/github/pmcbioc/pmc_result.duckdb]</span></span></span>
<span><span class="co">#&gt;       id pmcid    title                                      journal  year pmid </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     1 10484053 Plasma-Derived Small Extracellular Vesicl… Invest…  <span style="text-decoration: underline;">2</span>023 3767…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     2 10483746 A methylation- and immune-related lncRNA … Journa…  <span style="text-decoration: underline;">2</span>023 3767…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     3 10482981 A fungal sesquiterpene biosynthesis gene … Nature…  <span style="text-decoration: underline;">2</span>023 3767…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>     4 10482899 The mutational landscape of the adult hea… Nature…  <span style="text-decoration: underline;">2</span>023 3767…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>     5 10482349 Shigellainduces epigenetic reprogramming … Scienc…  <span style="text-decoration: underline;">2</span>023 <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     6 10478435 Sex differences in muscle protein express… Biolog…  <span style="text-decoration: underline;">2</span>023 3767…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>     7 10478242 Increased paternal corticosterone exposur… BMC Bi…  <span style="text-decoration: underline;">2</span>023 3766…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>     8 10478393 Effect of asciminib and vitamin K2 on Abe… BMC Ca…  <span style="text-decoration: underline;">2</span>023 3767…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>     9 10478213 Proteomic analysis unveils Gb3-independen… Journa…  <span style="text-decoration: underline;">2</span>023 3767…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>    10 10480445 Single cell multiomic analysis reveals di… Nature…  <span style="text-decoration: underline;">2</span>023 3766…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ more rows</span></span></span>
<span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"article"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">count</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   SQL [1 x 1]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB 0.8.1 [root@Darwin 21.6.0:R 4.3.1/~/a/github/pmcbioc/pmc_result.duckdb]</span></span></span>
<span><span class="co">#&gt;       n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> <span style="text-decoration: underline;">71</span>259</span></span></code></pre></div>
<p>Each row represents a result returned by the original PubMedCentral
query. The fields are</p>
<ul>
<li>
<code>id</code> is an internal identifier, representing the sequence
of the article in the original XML file. It is used in conjunction with
the <code>index</code> table.</li>
<li>
<code>pmcid</code> (PubMedCentral identifier), <code>title</code>,
<code>journal</code>, and <code>pmid</code> (PubMed identifier) are
self-explanatory.</li>
<li>
<code>year</code> is the earliest year recorded in the PubMedCentral
record. Some articles have multiple publication dates associated with,
e.g., electronic publication, submission to NIH, etc.</li>
</ul>
<p>Missing values occur when the XPath used to extract the data failed
to identify relevant informaton. I believe that the content of the XML
associated with each record is partly at the discretion of the
journal.</p>
<p>Use the <code>article</code> table to plot citations as a function of
year. Start by loading the <a href="https://cran.r-project.org/package=ggplot2" class="external-link">ggplot2</a>
package</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<p>Summarize the number of publications per year. Remember to
<code>collect()</code> the result so that the tibble is fully
in-memory.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pubs_per_year</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"article"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">count</span><span class="op">(</span><span class="va">year</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">collect</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Plot the relationship. One row is removed, corresponding to records
for which a year was not extracted from the XML. The final year is
incomplete, and there is a lag between publication and indexing by
PubMedCentral, so the last two points are an anomally.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">pubs_per_year</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">year</span>, y <span class="op">=</span> <span class="va">n</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 1 rows containing missing values (`geom_point()`).</span></span></code></pre></div>
<p><img src="introduction_files/figure-html/pubs-per-year-plot-1.png" width="700"></p>
<p>Journals with more than 100 articles mentioning Bioconductor are
summarized as (the unnamed journal corresponds to records for which the
XPath did not extract a name).</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"article"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">count</span><span class="op">(</span><span class="va">journal</span>, sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">n</span> <span class="op">&gt;=</span> <span class="fl">100</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">collect</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-8a104f5150b907b4b6b5" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-8a104f5150b907b4b6b5">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132"],["PLoS ONE","Scientific Reports","Nature Communications","BMC Genomics",null,"International Journal of Molecular Sciences","Nucleic Acids Research","Frontiers in Genetics","Frontiers in Immunology","BMC Bioinformatics","eLife","Oncotarget","Bioinformatics","Cancers","Frontiers in Plant Science","Proceedings of the National Academy of Sciences of the United States of America","Frontiers in Oncology","Frontiers in Microbiology","PLoS Genetics","Genome Biology","iScience","Genes","Nature","Cells","Cell reports","PeerJ","Frontiers in Cell and Developmental Biology","Plant Physiology","Genome Research","Clinical Epigenetics","Oncology Letters","Science Advances","BMC Cancer","Aging (Albany NY)","Cell","PLoS Computational Biology","bioRxiv","Communications Biology","BMC Plant Biology","BioMed Research International","JCI Insight","Epigenetics","The Journal of Clinical Investigation","Molecular Medicine Reports","PLoS Pathogens","BMC Medical Genomics","The Journal of Biological Chemistry","Frontiers in Pharmacology","G3: Genes|Genomes|Genetics","Cell Death &amp;Disease","The Plant Cell","The Journal of Experimental Medicine","Cell Reports","Nature genetics","mBio","Human Molecular Genetics","Genome Medicine","Journal of Translational Medicine","Cancer research","Genome Biology and Evolution","Development (Cambridge, England)","The EMBO Journal","Briefings in Bioinformatics","Applied and Environmental Microbiology","Frontiers in Physiology","Medicine","Plants","Oncogene","Genes &amp;Development","F1000Research","Molecular &amp;Cellular Proteomics : MCP","Clinical cancer research : an official journal of the American Association for Cancer Research","Molecular Oncology","Translational Psychiatry","Computational and Structural Biotechnology Journal","Blood","Journal of Virology","Frontiers in Cellular and Infection Microbiology","Microorganisms","Molecular Biology and Evolution","British Journal of Cancer","Molecular cell","Journal of Experimental Botany","The Journal of Neuroscience","Frontiers in Molecular Biosciences","Journal of immunology (Baltimore, Md. : 1950)","Scientific Data","PLoS Biology","BMC Biology","Frontiers in Endocrinology","Cell Reports Medicine","Cancer Cell International","Genetics","Experimental and Therapeutic Medicine","Journal of Cellular and Molecular Medicine","Biomedicines","Viruses","Molecular and Cellular Biology","EBioMedicine","Immunity","Biomolecules","EMBO Reports","Animals : an Open Access Journal from MDPI","STAR Protocols","Molecular Systems Biology","BMC Systems Biology","Nature medicine","Annals of Translational Medicine","Cancer Medicine","RNA","American Journal of Human Genetics","Epigenetics &amp;Chromatin","Breast Cancer Research : BCR","Nature immunology","Metabolites","Genomics Data","GigaScience","mSystems","Frontiers in Molecular Neuroscience","EMBO Molecular Medicine","Life Science Alliance","Nutrients","Molecular Therapy. Nucleic Acids","Microbiome","Journal for Immunotherapy of Cancer","PLOS ONE","Frontiers in Neuroscience","PLoS Neglected Tropical Diseases","Cancer cell","Molecular Metabolism","Biology","OncoTargets and therapy"],[3683,3386,2582,1924,1728,1546,1335,1208,1064,939,846,791,747,738,731,725,643,630,627,626,555,549,436,413,412,397,377,371,370,367,359,349,344,344,340,330,323,320,309,306,299,299,296,289,283,278,275,255,248,246,244,237,231,211,209,203,202,198,197,190,187,186,186,185,181,180,177,175,175,173,169,168,162,161,160,159,156,154,153,153,152,150,149,149,147,147,146,146,143,143,143,142,142,141,138,136,133,133,133,132,131,130,129,128,128,128,127,124,122,122,121,119,119,118,115,115,112,112,111,111,110,110,110,108,107,106,106,105,104,103,100,100]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>journal<\/th>\n      <th>n<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level4">
<h4 id="author">
<code>author</code><a class="anchor" aria-label="anchor" href="#author"></a>
</h4>
<p>The <code>author</code> table associates author surname and given
names with PubMedCentral identifiers. In some instances individual
authors were not extracted (e.g., because the journal reported the
author using a format the XPath did not capture). These entries are
<em>not</em> reported as <code>NA</code>, but are simply omitted from
the table. Thus some authors do not recieve credit for all their
publications.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"author"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   table&lt;author&gt; [?? x 3]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB 0.8.1 [root@Darwin 21.6.0:R 4.3.1/~/a/github/pmcbioc/pmc_result.duckdb]</span></span></span>
<span><span class="co">#&gt;    pmcid    surname givenname</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 10484053 Li      Bing     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 10484053 Sun     Nan      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 10484053 Yang    Fuhua    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 10484053 Guo     Kailei   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 10484053 Wu      Lingzi   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 10484053 Ma      Mingming </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 10484053 Shao    Hui      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 10484053 Li      Xiaorong </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 10484053 Zhang   Xiaomin  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 10483746 Chen    Lu       </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ more rows</span></span></span>
<span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"author"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">count</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   SQL [1 x 1]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB 0.8.1 [root@Darwin 21.6.0:R 4.3.1/~/a/github/pmcbioc/pmc_result.duckdb]</span></span></span>
<span><span class="co">#&gt;        n</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> <span style="text-decoration: underline;">733</span>519</span></span></code></pre></div>
<p>The most prolific authors (or at least unique combinations of surname
and given names) are found as:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"author"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">count</span><span class="op">(</span><span class="va">surname</span>, <span class="va">givenname</span>, sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">collect</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-58b78e6d07fc30a2964c" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-58b78e6d07fc30a2964c">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100"],["Wang","Zhang","Wang","Li","Zhang","Li","Wang","Wang","Li","Liu","Zhang","Li","Zhang","Zhang","Chen","Wang","Chen","Li","Wang","Zhang","Li","Zhang","Wang","Zhang","Liu","Wang","Zhang","Quackenbush","Wang","Liu","Huber","Zhang","Wang","Wang","Li","Zhang","Zhang","Wang","Wang","Coppola","Yan","Wu","Zhang","Wang","Haibe-Kains","Esteller","Liu","Barsh","Zhao","Wu","Zhang","Wang","Wang","Wang","Chen","Zhang","Medvedovic","Yang","Bock","Chen","Beck","Chen","Wang","Liu","Pfister","Liu","Takabe","Li","Li","Li","Li","Li","Chen","Zhang","Li","Horvath","Liu","Zhu","Wang","Carninci","Zhang","Zhang","Wang","Zhang","Caldas","Wang","Wang","Győrffy","Chen","Greally","Wang","Bammler","Zheng","Liu","Li","Liu","Wang","Robinson","Regev","Bickmore"],["Wei","Wei","Jing","Wei","Bin","Yan","Yan","Jun","Li","Yang","Yan","Jing","Yu","Yi","Wei","Yue","Xi","Yang","Xin","Li","Jun","Lei","Li","Jie","Wei","Ying","Hui","John","Tao","Yunlong","Wolfgang","Jing","Jie","Kai","Xin","Jian","Ying","Hui","Lu","Giovanni","Li","Xiwei","Bo","Lei","Benjamin","Manel","Ying","Gregory S.","Wei","Hao","Yong","Ting","Yu","Qi","Jing","Hao","Mario","Yang","Christoph","Xin","Stephan","Li","Yi","Song","Stefan M.","Li","Kazuaki","Jie","Bo","Lin","Hui","Ying","Gang","Qi","Na","Steve","Jing","Jun","Hong","Piero","Chi","Lin","Jian","Bing","Carlos","Fang","Dan","Balázs","Hao","John M.","Qian","Theo K.","Wei","Jun","Lei","Lin","Hao","Mark D.","Aviv","Wendy A."],[179,166,161,159,121,121,120,119,118,116,112,111,111,107,107,105,103,100,97,97,97,95,94,91,90,88,87,84,83,82,82,82,81,81,81,81,80,80,79,77,76,76,75,75,73,73,72,72,72,71,71,70,70,70,68,68,67,67,66,66,65,65,65,65,64,64,64,63,63,62,62,62,62,61,61,61,61,61,60,60,60,60,59,59,59,59,59,58,58,58,58,58,58,58,57,57,57,57,57,56]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>surname<\/th>\n      <th>givenname<\/th>\n      <th>n<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>Data inconsistencies are not easily resolved, e.g., are these all
me?</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"author"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">surname</span> <span class="op">==</span> <span class="st">"Morgan"</span>, <span class="va">givenname</span> <span class="op">%like%</span> <span class="st">"Martin%"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">count</span><span class="op">(</span><span class="va">surname</span>, <span class="va">givenname</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   SQL [2 x 3]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB 0.8.1 [root@Darwin 21.6.0:R 4.3.1/~/a/github/pmcbioc/pmc_result.duckdb]</span></span></span>
<span><span class="co">#&gt;   surname givenname     n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Morgan  Martin       26</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Morgan  Martin T.     5</span></span></code></pre></div>
<p>Publications of individual authors can be found by joining the author
and article tables using PubMedCentral id.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"author"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">surname</span> <span class="op">==</span> <span class="st">"Morgan"</span>, <span class="va">givenname</span> <span class="op">%like%</span> <span class="st">"Martin%"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"article"</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"pmcid"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">collect</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span></span>
<span>        author <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">surname</span>, <span class="va">givenname</span>, sep <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>,</span>
<span>        <span class="co">## remove "\n" in one record</span></span>
<span>        title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"[[:space:]]+"</span>, <span class="st">" "</span>, <span class="va">title</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="va">author</span>, <span class="va">year</span>, <span class="va">title</span>, <span class="va">journal</span>, <span class="va">pmcid</span>, <span class="va">pmid</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-cb2d2f90655d01e6cb15" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-cb2d2f90655d01e6cb15">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31"],["Morgan, Martin","Morgan, Martin","Morgan, Martin T.","Morgan, Martin T.","Morgan, Martin","Morgan, Martin T.","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin","Morgan, Martin T.","Morgan, Martin","Morgan, Martin","Morgan, Martin T."],[2015,2014,2013,2012,2010,2011,2009,2014,2013,2012,2021,2020,2020,2020,2020,2020,2020,2018,2018,2016,2015,2023,2023,2022,2021,2020,2019,2019,2019,2018,2018],["Orchestrating high-throughput genomic analysis with Bioconductor","Scalable Genomics with R and Bioconductor","Software for Computing and Annotating Genomic Ranges","Bacterial Communities in Women with Bacterial Vaginosis: High Resolution Phylogenetic Analyses Reveal Relationships of Microbiota to Clinical Criteria","Genomic Differences Between Estrogen Receptor (ER)-Positive and ER-Negative Human Breast Carcinoma Identified by Single Nucleotide Polymorphism Array Comparative Genome Hybridization Analysis","Genome-wide MyoD binding in skeletal muscle cells: a potential for broad cellular reprogramming","ShortRead: a bioconductor package for input, quality assessment and exploration of high-throughput sequence data","VariantAnnotation: a Bioconductorpackage for exploration and annotation of genetic variants","Gene expression changes in adipose tissue with diet- and/or exercise-induced weight loss","MYC-driven tumorigenesis is inhibited by WRN syndrome gene deficiency","Bioconductortoolchain for reproducible bioinformatics pipelines using Rcwland RcwlPipelines","Multiomic Integration of Public Oncology Databases in Bioconductor","Multi-omic analysis of subtype evolution and heterogeneity in high-grade serous ovarian carcinoma","Cancer Moonshot Immuno-Oncology Translational Network (IOTN): accelerating the clinical translation of basic discoveries for improving immunotherapy and immunoprevention of cancer","Reliable Analysis of Clinical Tumor-Only Whole-Exome Sequencing Data","Toward a gold standard for benchmarking gene set enrichment analysis","Global Alliance for Genomics and Health Meets Bioconductor: Toward Reproducible and Agile Cancer Genomics at Cloud Scale","Software for the integration of multi-omics experiments in Bioconductor","Accessible, curated metagenomic data through ExperimentHub","The Bioconductor channel in F1000Research","Public data and open source tools for multi-assay genomic investigation of disease","Evaluation of software impact designed for biomedical research: Are we measuring what’s meaningful?","RaggedExperiment: the missing link between genomic ranges and matrices in Bioconductor","Inverting the model of genomics data sharing with the NHGRI Genomic Data Science Analysis, Visualization, and Informatics Lab-space","Open-source Software Sustainability Models: Initial White Paper From the Informatics Technology for Cancer Research Sustainability and Industry Partnership Working Group","Tximeta: Reference sequence checksums for provenance identification in RNA-seq","Orchestrating Single-Cell Analysis with Bioconductor","BiocPkgTools: Toolkit for mining the Bioconductorpackage ecosystem","restfulSE: A semantically rich interface for cloud-scale genomics with Bioconductor","gwasurvivr: an R package for genome-wide survival analysis","Orchestrating a community-developed computational workshop and accompanying training materials"],["Nature methods","Statistical science : a review journal of the Institute of Mathematical Statistics","PLoS Computational Biology","PLoS ONE","Cancer",null,"Bioinformatics","Bioinformatics","Cancer prevention research (Philadelphia, Pa.)","Molecular cancer research : MCR","Bioinformatics","JCO Clinical Cancer Informatics","Cancer research","Journal for Immunotherapy of Cancer","JCO Clinical Cancer Informatics","Briefings in Bioinformatics","JCO Clinical Cancer Informatics","Cancer research","Nature methods","F1000Research","Briefings in Bioinformatics","ArXiv","Bioinformatics","Cell Genomics","Journal of Medical Internet Research","PLoS Computational Biology","Nature methods","F1000Research","F1000Research","Bioinformatics","F1000Research"],["4509590","5181792","3738458","3377712","4521590","2910615","2752612","4080743","3738189","3707802","8504628","7608653","7572645","7304845","7259876","7820859","7265787","5679241","5862039","4786899","4945830","10274942","10272705","8863334","8686402","7059966","7358058","6584971","6392152","7963072","6234736"],["25633503","28018047","23950696","22719852","21523713","20412780","19654119","24681907","23341572","22301954","33772584","33119407","32747365","32554617","32282230","32026945","32453635","29092936","29088129","26998224","26463000","37332562","37208161","35199087","34860667","32097405","31792435","31249680","30828438","30395168","30473781"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>author<\/th>\n      <th>year<\/th>\n      <th>title<\/th>\n      <th>journal<\/th>\n      <th>pmcid<\/th>\n      <th>pmid<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level4">
<h4 id="keyword-and-refpmid">
<code>keyword</code> and <code>refpmid</code><a class="anchor" aria-label="anchor" href="#keyword-and-refpmid"></a>
</h4>
<p>The <code>keyword</code> table summarizes keywords associated with
each publication. Keywords are not standardized, e.g,
<code>bioinformatic</code> and <code>bioinformatics</code> are both
used. Nonetheless, this provides some insight into areas where
Bioconductor is useful; an analysis could associate keyword with year to
model shifting areas of interest.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"keyword"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   table&lt;keyword&gt; [?? x 2]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB 0.8.1 [root@Darwin 21.6.0:R 4.3.1/~/a/github/pmcbioc/pmc_result.duckdb]</span></span></span>
<span><span class="co">#&gt;    pmcid    keyword                     </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 10484053 CXCL5                       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 10484053 exosomes                    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 10484053 miR-410-3p                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 10484053 small extracellular vesicles</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 10484053 Vogt-Koyanagi-Harada        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 10483746 mrlncRNA                    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 10483746 irlncRNA                    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 10483746 OC                          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 10483746 Chemoresistance             </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 10483746 Tumor-infiltrating          </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ more rows</span></span></span>
<span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"keyword"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">count</span><span class="op">(</span>keyword <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="va">keyword</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">n</span> <span class="op">&gt;=</span> <span class="fl">100</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">collect</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-977cb8d816847155acbf" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-977cb8d816847155acbf">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255"],["gene expression","rna-seq","dna methylation","transcriptome","transcriptomics","epigenetics","prognosis","breast cancer","bioinformatics","cancer","microarray","inflammation","microrna","biomarker","biomarkers","rna sequencing","proteomics","mirna","machine learning","immunotherapy","colorectal cancer","genomics","differentially expressed genes","mouse","aging","immunology","prostate cancer","tumor microenvironment","metabolomics","gene regulation","genetics","transcription","metabolism","hepatocellular carcinoma","molecular biology","microbiome","systems biology","metastasis","human","cancer genomics","lncrna","apoptosis","tcga","covid-19","methylation","computational biology and bioinformatics","cell biology","sars-cov-2","obesity","bioinformatics analysis","transcription factor","alternative splicing","rnaseq","chromatin","glioblastoma","gastric cancer","mitochondria","ovarian cancer","lung adenocarcinoma","differentiation","melanoma","development","next-generation sequencing","data integration","gene expression profiling","micrornas","immune infiltration","epigenomics","transcription factors","oxidative stress","rna-sequencing","wgcna","cell cycle","stem cells","meta-analysis","innate immunity","pancreatic cancer","survival","zebrafish","differential expression","mass spectrometry","hypoxia","immune response","lung cancer","autophagy","oncology","proliferation","evolution","macrophages","alzheimer’s disease","neuroscience","chip-seq","mrna","long non-coding rna","microglia","colon cancer","gwas","bladder cancer","gene ontology","transcriptome analysis","liver","mirnas","drosophila","macrophage","pathway analysis","cytokines","transcriptional regulation","differential gene expression","software","gene regulatory networks","glioma","microbiota","diabetes","angiogenesis","sequencing","extracellular vesicles","microarrays","omics","placenta","scrna-seq","t cells","senescence","immunity","diagnosis","pregnancy","neurodegeneration","rna","multi-omics","epigenetic","network analysis","brain","arabidopsis","next generation sequencing","network pharmacology","skeletal muscle","p53","signal transduction","gut microbiota","developmental biology","microbiology","infection","molecular medicine","functional genomics","chemotherapy","fibrosis","overall survival","hub genes","survival analysis","metagenomics","invasion","single-cell rna sequencing","drug resistance","schizophrenia","osteosarcoma","hippocampus","cancer stem cells","extracellular matrix","translation","autoimmunity","lipid metabolism","exosomes","metabolome","prognostic model","comparative genomics","asthma","data mining","biological sciences","emt","proteome","computational biology","microbial ecology","the cancer genome atlas","immune","ageing","type 2 diabetes","prognostic markers","network","pathogenesis","resistance","neuroinflammation","atac-seq","embryonic stem cells","signature","cerna","microarray analysis","nomogram","atherosclerosis","migration","cancer models","single-cell rna-seq","normalization","long non-coding rnas","pig","diseases","gene expression analysis","regeneration","alzheimer's disease","cancer microenvironment","other","single cell","inflammatory bowel disease","interferon","enhancer","prognostic signature","high-throughput sequencing","clustering","precision medicine","cervical cancer","pathway","tumour immunology","arabidopsis thaliana","heart failure","cancer immunotherapy","maize","protein-protein interaction network","lncrnas","long noncoding rna","ferroptosis","gene signature","virulence","circrna","non-small cell lung cancer","sepsis","genome","abiotic stress","oncogenes","weighted gene co-expression network analysis","stress","differentially expressed gene","parkinson’s disease","circular rna","monocytes","acute myeloid leukemia","chromatin accessibility","plant sciences","rheumatoid arthritis","classification","pluripotency","immune microenvironment","multiple sclerosis","histone modification","tumour biomarkers","neuroblastoma","tuberculosis","osteoarthritis","personalized medicine","mutation","mesenchymal stem cells","viral infection","gene regulatory network","tumor immune microenvironment","bioinformatic analysis","geo","flow cytometry","chicken"],[2536,2243,2087,2032,1901,1419,1394,1314,1274,1200,1001,884,870,852,824,816,761,637,636,621,604,579,570,562,534,533,531,525,520,510,509,504,504,503,497,464,450,441,432,423,422,420,405,397,397,386,385,369,364,348,333,333,329,328,326,318,317,314,313,311,301,295,293,291,289,288,282,280,278,278,277,277,268,262,262,260,260,258,258,255,254,254,252,251,243,243,243,236,236,234,233,224,219,217,210,208,207,203,203,200,199,196,196,192,191,190,188,186,184,182,182,181,181,180,180,178,176,175,174,172,169,169,168,167,166,164,163,163,163,158,158,157,156,155,155,155,154,153,152,152,151,150,150,148,148,148,147,147,147,146,146,146,146,145,145,142,142,140,140,139,139,137,136,135,134,134,134,133,131,131,131,130,130,129,129,129,128,128,128,128,127,127,126,126,126,126,125,125,124,123,123,123,123,123,122,122,122,122,121,120,120,120,120,120,119,119,119,119,119,119,118,118,117,117,116,116,115,114,114,113,112,112,111,111,111,111,110,110,109,108,108,108,108,107,107,107,107,105,105,104,103,103,103,103,102,102,102,102,101,101,101,101,100,100,100]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>keyword<\/th>\n      <th>n<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>The <code>refpmid</code> table summarizes the PubMed identifiers of
citations in each article.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"refpmid"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   table&lt;refpmid&gt; [?? x 2]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB 0.8.1 [root@Darwin 21.6.0:R 4.3.1/~/a/github/pmcbioc/pmc_result.duckdb]</span></span></span>
<span><span class="co">#&gt;    pmcid    refpmid </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 10484053 20036008</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 10484053 29138505</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 10484053 25108386</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 10484053 29321438</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 10484053 29111815</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 10484053 29998211</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 10484053 23584393</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 10484053 28955186</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 10484053 21324974</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 10484053 15741221</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ more rows</span></span></span></code></pre></div>
<p>One can ask which publications mentioning ‘Bioconductor’ are most
cited by other publications mentioning ‘Bioconductor’</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## FIXME: is this correct?</span></span>
<span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"refpmid"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">count</span><span class="op">(</span><span class="va">refpmid</span>, sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">right_join</span><span class="op">(</span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"article"</span><span class="op">)</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>refpmid <span class="op">=</span> <span class="st">"pmid"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">collect</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: ORDER BY is ignored in subqueries without LIMIT</span></span>
<span><span class="co">#&gt; <span style="color: #0000BB;">ℹ</span> Do you need to move arrange() later in the pipeline or use window_order() instead?</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e64792f0f0b3e9f177ff" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e64792f0f0b3e9f177ff">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100"],["19910308","25516281","25605792","15461798","25260700","22455463","20979621","20196867","21816040","19114008","22287627","24478339","19617889","25633503","26704973","24485249","22383036","23193258","31178118","25822800","20644199","23950696","25352553","22257669","20132535","23845962","12582260","23323831","23740750","24113773","27560171","23558742","23630581","27214047","27924014","26925227","24336805","20427518","34557778","20688976","27909575","16284200","30341162","21715386","19106121","25150836","30783653","23175756","23222703","29628290","21118553","29409532","28991892","21085593","20459804","21414208","23177740","26653891","20129251","28088763","33237311","22722343","20167110","26813401","31740819","24670763","31870423","25599564","19473525","24336642","32416070","27508062","22703947","27765066","21903743","28825705","29195078","23476028","23631413","26457759","27717381","31341288","23034086","19654119","28575171","24670764","27508061","20738864","24495553","25972926","31068700","29344893","29539639","22848472","29608177","19468054","17099226","27122128","19295514","28035024"],[13587,7279,5460,4278,3100,2999,2104,1644,1569,1546,1448,1397,1281,1208,1187,1083,1037,984,961,954,864,855,806,793,745,716,690,667,643,630,572,536,529,519,519,514,463,433,431,425,421,393,389,389,374,363,343,334,328,326,319,310,303,300,295,286,283,273,271,270,268,264,257,256,254,247,243,243,242,241,237,236,231,230,229,221,220,216,214,211,209,209,207,204,204,202,201,197,196,193,190,189,188,186,185,181,172,172,170,168],[68869,59385,59026,71208,59948,65451,67699,68515,66623,69683,65870,61645,69160,58919,55834,61585,65707,64348,38851,58434,67890,62861,59718,65925,68589,63045,71256,64055,63270,62423,53449,63608,63509,54296,52709,55167,61953,68349,21573,67987,52592,71076,42494,66778,69688,60195,40725,64357,64275,45612,67586,46477,48181,67640,68298,67175,64366,55957,68654,51671,27598,64598,68554,55526,35906,61238,35184,59415,69307,61966,32216,52560,65211,52679,66152,48840,47408,63756,63448,56587,52803,37807,64581,69133,49926,61239,53535,67920,61572,58963,39374,47000,45924,65006,45675,69313,70856,54586,69498,52226],["2796818","4302049","4402510","545600","4287950","3339379","3218662","2864565","3163565","2631488","3378882","4016708","3159387","4509590","4856967","4053721","3334321","3531084","6687398","4739640","2928508","3738458","4383874","3307112","2872874","3875132","150247","3618321","3702256","3826632","5032908","3664803","3632530","4927377","5210637","4712774","3928520","2881355","8454663","2944196","5112579","1283542","6211645","3125809","2639074","4404308","6486549","3546795","3869392","5982584","3012676","5802054","5937676","2981572","3098059","3068975","3780611","4676162","2818769","5408845","7779004","3460195","2838869","4728800","6884693","5215096","6927181","4283580","2696452","3904520","7227586","4955027","3446316","5073889","3227109","5764547","5990023","3627582","3769145","4636487","5055731","7015180","3491415","2752612","5870550","4529748","4934518","2945788","4053810","4429355","6697103","5895181","6093218","3405143","6152897","2705236","1669752","4848819","2910248","5408810"],["edgeR: a Bioconductor package for differential expression analysis of digital gene expression data","Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2","limmapowers differential expression analyses for RNA-sequencing and microarray studies","Bioconductor: open software development for computational biology and bioinformatics","HTSeq—a Python framework to work with high-throughput sequencing data","clusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters","Differential expression analysis for sequence count data","A scaling normalization method for differential expression analysis of RNA-seq data","RSEM: accurate transcript quantification from RNA-Seq data with or without a reference genome","WGCNA: an R package for weighted correlation network analysis","Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation","Minfi: a flexible and comprehensive Bioconductor package for the analysis of Infinium DNA methylation microarrays","Mapping Identifiers for the Integration of Genomic Datasets with the R/Bioconductor package biomaRt","Orchestrating high-throughput genomic analysis with Bioconductor","TCGAbiolinks: an R/Bioconductor package for integrative analysis of TCGA data","voom: precision weights unlock linear model analysis tools for RNA-seq read counts","Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks","NCBI GEO: archive for functional genomics data sets—update","Comprehensive integration of single-cell data","Robust enumeration of cell subsets from tissue expression profiles","The Genome Analysis Toolkit: A MapReduce framework for analyzing next-generation DNA sequencing data","Software for Computing and Annotating Genomic Ranges","STRING v10: protein–protein interaction networks, integrated over the tree of life","The svapackage for removing batch effects and other unwanted variation in high-throughput experiments","Gene ontology analysis for RNA-seq: accounting for selection bias","De novotranscript sequence reconstruction from RNA-Seq: reference generation and analysis with Trinity","Summaries of Affymetrix GeneChip probe level data","GSVA: gene set variation analysis for microarray and RNA-Seq data","Pathview: an R/Bioconductor package for pathway-based data integration and visualization","Inferring tumour purity and stromal and immune cell admixture from expression data","Transcript-level expression analysis of RNA-seq experiments with HISAT, StringTie, and Ballgown","The Subread aligner: fast, accurate and scalable read mapping by seed-and-vote","phyloseq: An R Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data","DADA2: High resolution sample inference from Illumina amplicon data","The STRING database in 2017: quality-controlled protein–protein association networks, made broadly accessible","Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences","Causal analysis approaches in Ingenuity Pathway Analysis","ConsensusClusterPlus: a class discovery tool with confidence assessments and item tracking","clusterProfiler 4.0: A universal enrichment tool for interpreting omics data","A framework for oligonucleotide microarray preprocessing","A step-by-step workflow for low-level analysis of single-cell RNA-seq data with Bioconductor","Evolving gene/transcript definitions significantly alter the interpretation of GeneChip data","Maftools: efficient and comprehensive analysis of somatic variants in cancer","KOBAS 2.0: a web server for annotation and identification of enriched pathways and diseases","arrayQualityMetrics—a bioconductor package for quality assessment of microarray data","Normalization of RNA-seq data using factor analysis of control genes or samples","The R package Rsubreadis easier, faster, cheaper and better for alignment and quantification of RNA sequencing reads","A beta-mixture quantile normalization method for correcting probe design bias in Illumina Infinium 450 k DNA methylation data","Differential analysis of gene regulation at transcript resolution with RNA-seq","The Immune Landscape of Cancer","Comparison of Beta-value and M-value methods for quantifying methylation levels by microarray analysis","SCANPY: large-scale single-cell gene expression data analysis","SCENIC: Single-cell regulatory network inference and clustering","Enrichment Map: A Network-Based Method for Gene-Set Enrichment Visualization and Interpretation","ChIPpeakAnno: a Bioconductor package to annotate ChIP-seq and ChIP-chip data","pROC: an open-source package for R and S+ to analyze and compare ROC curves","Genome-wide Methylation Profiles Reveal Quantitative Views of Human Aging Rates","MAST: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell RNA sequencing data","An integrated genomic analysis identifies clinically relevant subtypes of glioblastoma characterized by abnormalities in PDGFRA, IDH1, EGFR and NF1","Scater: pre-processing, quality control, normalization and visualization of single-cell RNA-seq data in ","The STRING database in 2021: customizable protein–protein networks, and functional characterization of user-uploaded gene/measurement sets","Detecting differential usage of exons from RNA-seq data","Evaluation of statistical methods for normalization and differential expression in mRNA-Seq experiments","A survey of best practices for RNA-seq data analysis","Fast, sensitive, and accurate integration of single cell data with Harmony","An atlas of active enhancers across human cell types and tissues","Normalization and variance stabilization of single-cell RNA-seq data using regularized negative binomial regression","Functional normalization of 450k methylation array data improves replication in large cancer studies","GAGE: generally applicable gene set enrichment for pathway analysis","ChAMP: 450k Chip Analysis Methylation Pipeline","Imbalanced Host Response to SARS-CoV-2 Drives Development of COVID-19","Bioconductor Workflow for Microbiome Data Analysis: from raw reads to community analyses","SWAN: Subset-quantile Within Array Normalization for Illumina Infinium HumanMethylation450 BeadChips","Estimating the population abundance of tissue-infiltrating immune and stromal cell populations using gene expression","Differential expression in RNA-seq: A matter of depth","Reversed graph embedding resolves complex single-cell trajectories","A Next Generation Connectivity Map: L1000 platform and the first 1,000,000 profiles","Low-level processing of Illumina Infinium DNA Methylation BeadArrays","A data-driven approach to preprocessing Illumina 450K methylation array data","The Consensus Molecular Subtypes of Colorectal Cancer","Critical evaluation of the Illumina MethylationEPIC BeadChip microarray for whole-genome DNA methylation profiling","Reproducible, interactive, scalable and extensible microbiome data science using QIIME ","methylKit: a comprehensive R package for the analysis of genome-wide DNA methylation profiles","ShortRead: a bioconductor package for input, quality assessment and exploration of high-throughput sequence data","karyoploteR: an R/Bioconductor package to plot customizable genomes displaying arbitrary data","A promoter-level mammalian expression atlas","From reads to genes to pathways: differential expression analysis of RNA-Seq experiments using Rsubread and the edgeR quasi-likelihood pipeline","Galaxy: a comprehensive approach for supporting accessible, reproducible, and transparent computational research in the life sciences","Accounting for cellular heterogeneity is critical in epigenome-wide association studies","De novoidentification of differentially methylated regions in the human genome","Next-generation characterization of the Cancer Cell Line Encyclopedia","Profiling tumor infiltrating immune cells with CIBERSORT","DNA methylation-based classification of central nervous system tumours","Differential DNA Methylation in Purified Human Blood Cells: Implications for Cell Lineage and Studies on Disease Susceptibility","Batch effects in single-cell RNA sequencing data are corrected by matching mutual nearest neighbours","rtracklayer: an R package for interfacing with genome browsers","NCBI GEO: mining tens of millions of expression profiles—database and tools update","Pooling across cells to normalize single-cell RNA sequencing data with many zero counts","Histone Modifications at Human Enhancers Reflect Global Cell Type-Specific Gene Expression","Preprocessing, normalization and integration of the Illumina HumanMethylationEPIC array with minfi"],["Bioinformatics","Genome Biology","Nucleic Acids Research",null,"Bioinformatics","OMICS : a Journal of Integrative Biology","Genome Biology","Genome Biology","BMC Bioinformatics",null,"Nucleic Acids Research","Bioinformatics","Nature protocols","Nature methods","Nucleic Acids Research","Genome Biology","Nature Protocols","Nucleic Acids Research","Cell","Nature methods","Genome Research","PLoS Computational Biology","Nucleic Acids Research","Bioinformatics","Genome Biology","Nature protocols",null,"BMC Bioinformatics","Bioinformatics","Nature Communications","Nature protocols","Nucleic Acids Research","PLoS ONE","Nature methods","Nucleic Acids Research","F1000Research","Bioinformatics","Bioinformatics","The Innovation","Bioinformatics","F1000Research",null,"Genome Research","Nucleic Acids Research",null,"Nature biotechnology","Nucleic Acids Research","Bioinformatics","Nature biotechnology","Immunity","BMC Bioinformatics","Genome Biology","Nature methods","PLoS ONE","BMC Bioinformatics","BMC Bioinformatics","Molecular cell","Genome Biology",null,"Bioinformatics","Nucleic Acids Research","Genome Research","BMC Bioinformatics","Genome Biology","Nature methods","Nature","Genome Biology","Genome Biology",null,"Bioinformatics","Cell","F1000Research","Genome Biology","Genome Biology","Genome Research","Nature methods","Cell","Nucleic Acids Research","BMC Genomics","Nature medicine","Genome Biology","Nature biotechnology","Genome Biology","Bioinformatics","Bioinformatics","Nature","F1000Research","Genome Biology","Genome Biology","Epigenetics &amp;Chromatin","Nature","Methods in molecular biology (Clifton, N.J.)","Nature","PLoS ONE","Nature biotechnology","Bioinformatics","Nucleic Acids Research","Genome Biology",null,"Bioinformatics"],[2009,2014,2015,2004,2014,null,2010,2010,2011,2008,2012,2014,2009,2015,2015,2014,2012,2012,2019,2015,null,2013,2014,2012,2010,2013,null,2013,2013,2013,2016,2013,2013,2016,2016,2016,2013,2010,2021,2010,2016,2005,null,2011,2008,2014,2019,2012,2012,2018,2010,2018,2017,2010,2010,2011,2012,2015,2011,2017,2020,2012,2010,2016,2019,2017,2019,2014,2009,2013,2020,2016,2012,2016,2011,2017,2018,2013,2013,2015,2016,2020,2012,2009,2017,2015,2016,2010,2014,2015,2019,2019,2018,2012,2018,2009,2006,2016,2009,2016]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>refpmid<\/th>\n      <th>n<\/th>\n      <th>id<\/th>\n      <th>pmcid<\/th>\n      <th>title<\/th>\n      <th>journal<\/th>\n      <th>year<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="section level3">
<h3 id="querying-individual-records">Querying individual records<a class="anchor" aria-label="anchor" href="#querying-individual-records"></a>
</h3>
<p>The XML records are extremely rich. Individual records can be
retrieved and queried. For instance for the fifth record</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"article"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">id</span> <span class="op">==</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   SQL [1 x 6]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB 0.8.1 [root@Darwin 21.6.0:R 4.3.1/~/a/github/pmcbioc/pmc_result.duckdb]</span></span></span>
<span><span class="co">#&gt;      id pmcid    title                                       journal  year pmid </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>     5 10482349 Shigellainduces epigenetic reprogramming o… Scienc…  <span style="text-decoration: underline;">2</span>023 <span style="color: #BB0000;">NA</span></span></span></code></pre></div>
<p>we can use the <code>index</code> table to query the full XML record
using XPath to count the number references</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">xml_file</span> <span class="op">&lt;-</span> <span class="st">"~/a/github/pmcbioc/pmc_result.xml"</span></span>
<span><span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"index"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">id</span> <span class="op">==</span> <span class="fl">5</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/xml_xpath.html">xml_xpath</a></span><span class="op">(</span></span>
<span>        <span class="st">"count(//article/back/ref-list/ref)"</span>,</span>
<span>        xml_file <span class="op">=</span> <span class="va">xml_file</span></span>
<span>    <span class="op">)</span></span>
<span><span class="co">#&gt; [1] 81</span></span></code></pre></div>
<p>The full XML of the first five records is</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">first_five</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">tbl</span><span class="op">(</span><span class="va">db</span>, <span class="st">"index"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">id</span> <span class="op">&lt;=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/xml_xpath.html">xml_xpath</a></span><span class="op">(</span>xml_file <span class="op">=</span> <span class="va">xml_file</span><span class="op">)</span></span></code></pre></div>
<p>These records can be queried using, e.g, XPath or the
<code>xmlChildren()</code> and <code>[[</code> of the XML package.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">first_five</span><span class="op">[</span><span class="st">"//article-meta//article-title"</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">XML</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/XML/man/xmlValue.html" class="external-link">xmlValue</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] NA</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="final-steps-1">Final steps<a class="anchor" aria-label="anchor" href="#final-steps-1"></a>
</h2>
<p>Remember to disconnect from the database.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">db_disconnect</span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span><span class="co">#&gt; pmcbioc_db: ~/a/github/pmcbioc/pmc_result.duckdb</span></span>
<span><span class="co">#&gt; connected: FALSE</span></span>
<span><span class="co">#&gt; read_only: TRUE</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 Patched (2023-08-16 r84986)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin21.6.0 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Monterey 12.6.8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Users/ma38727/bin/R-4-3-branch/lib/libRblas.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Users/ma38727/bin/R-4-3-branch/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_3.4.3      pmcbioc_0.0.0.9000 dplyr_1.1.2       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] sass_0.4.7        utf8_1.2.3        generics_0.1.3    spdl_0.0.5       </span></span>
<span><span class="co">#&gt;  [5] stringi_1.7.12    digest_0.6.33     magrittr_2.0.3    evaluate_0.21    </span></span>
<span><span class="co">#&gt;  [9] grid_4.3.1        fastmap_1.1.1     blob_1.2.4        rprojroot_2.0.3  </span></span>
<span><span class="co">#&gt; [13] jsonlite_1.8.7    DBI_1.1.3         purrr_1.0.2       fansi_1.0.4      </span></span>
<span><span class="co">#&gt; [17] crosstalk_1.2.0   scales_1.2.1      XML_3.99-0.14     textshaping_0.3.6</span></span>
<span><span class="co">#&gt; [21] jquerylib_0.1.4   duckdb_0.8.1-2    cli_3.6.1         rlang_1.1.1      </span></span>
<span><span class="co">#&gt; [25] dbplyr_2.3.3      ellipsis_0.3.2    munsell_0.5.0     withr_2.5.0      </span></span>
<span><span class="co">#&gt; [29] cachem_1.0.8      yaml_2.3.7        tools_4.3.1       memoise_2.0.1    </span></span>
<span><span class="co">#&gt; [33] colorspace_2.1-0  DT_0.28           vctrs_0.6.3       R6_2.5.1         </span></span>
<span><span class="co">#&gt; [37] lifecycle_1.0.3   RcppSpdlog_0.0.14 stringr_1.5.0     htmlwidgets_1.6.2</span></span>
<span><span class="co">#&gt; [41] fs_1.6.3          ragg_1.2.5        pkgconfig_2.0.3   desc_1.4.2       </span></span>
<span><span class="co">#&gt; [45] pkgdown_2.0.7     pillar_1.9.0      bslib_0.5.1       gtable_0.3.4     </span></span>
<span><span class="co">#&gt; [49] glue_1.6.2        Rcpp_1.0.11       systemfonts_1.0.4 highr_0.10       </span></span>
<span><span class="co">#&gt; [53] xfun_0.40         tibble_3.2.1      tidyselect_1.2.0  knitr_1.43       </span></span>
<span><span class="co">#&gt; [57] farver_2.1.1      htmltools_0.5.6   labeling_0.4.2    rmarkdown_2.24   </span></span>
<span><span class="co">#&gt; [61] compiler_4.3.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martin Morgan.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
